lumen/sources/intake_sql.py

```diff
@@ -40,7 +39,7 @@ def _get_source(self, table):
         try:
             source = self.cat[table]
         except KeyError as e:
-            raise ValidationError(
+            raise KeyError(
```

Comment:
When I do `sources = [src for src in available_sources if table in src]` in [`TableAgent`](https://github.com/holoviz/lumen/blob/main/lumen/ai/agents.py#L422)

it crashes because
https://github.com/holoviz/lumen/blob/905ac36c791a8111aef90fa8593eef459b04a6ff/lumen/sources/intake_dremio.py#L41
`__contains__` expects a KeyError